src/arch/amdgpu/vega/insts/vop1.cc

```diff
@@ -433,7 +433,22 @@ namespace VegaISA
     void
     Inst_VOP1__V_CVT_F16_F32::execute(GPUDynInstPtr gpuDynInst)
     {
-        panicUnimplemented();
+        Wavefront *wf = gpuDynInst->wavefront();
+        ConstVecOperandF32 src(gpuDynInst, instData.SRC0);
+        VecOperandU32 vdst(gpuDynInst, instData.VDST);
+
+        src.readSrc();
+
+        for (int lane = 0; lane < NumVecElemPerVecReg; ++lane) {
+            if (wf->execMask(lane)) {
+                float tmp = src[lane];
+                AMDGPU::mxfloat16 out(tmp);
+
+                vdst[lane] = (out.data >> 16);
```

Comment:
Since the MXFP types are manipulated using a union of a struct (sign, exponent, mantissa, zero padding) the actual data of an N bit type is store in bits [31:(32-N)], hence the need for the shift.  Although it may be cleaner to just have a getter method which does the shift for you.